state.go

```diff
@@ -397,6 +397,16 @@ func NewUpdateErrorState(err menderError, update UpdateResponse) State {
 func (ue *UpdateErrorState) Handle(c Controller) (State, bool) {
 	// TODO error handling
 	c.ReportUpdateStatus(ue.update, statusFailure)
+
+	// TODO fetch logs from logger
+	c.UploadLog(ue.update, []LogEntry{
+		LogEntry{
+			Timestamp: time.Now().Format(time.RFC3339),
+			Level:     "error",
+			Message:   "update failed",
+		},
+	})
```

Comment:
Yeah, might be the same scenario as with update status report. Say, if we fail to report update failure, we should just keep on trying until we succeed. There's also a question what happens when there's a reboot in between. And there's obviously possibility of rolling back. IMO we should go over the scenarios with @eysteins seems like we can't/should not really try to solve it here.